fmn/api/handlers/admin.py

```diff
@@ -0,0 +1,68 @@
+import logging
+
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from ...database.model import Rule
+from ...messages.rule import RuleUpdateV1
+from .. import api_models
+from ..auth import Identity, get_identity_optional
+from ..database import gen_db_session
+from ..messaging import publish
+
+log = logging.getLogger(__name__)
+
+router = APIRouter(prefix="/admin")
+
+
+@router.get("", response_model=bool, tags=["users/rules"])
+async def get_is_admin(
+    identity: Identity = Depends(get_identity_optional),
+):
+    if identity and identity.admin:
+        return True
+    else:
+        return False
+
+
+@router.get("/rules", response_model=list[api_models.Rule], tags=["users/rules"])
+async def get_disabled_rules(
+    identity: Identity = Depends(get_identity_optional),
+    db_session: AsyncSession = Depends(gen_db_session),
+):
+    if not identity.admin:
```

Comment:
Instead of repeating this test in all the endpoints, I would suggest adding this to `fmn/api/auth.py`:

```py
async def get_identity_admin(identity: Identity = Depends(get_identity)):
    if not identity.admin:
        raise HTTPException(status.HTTP_403_FORBIDDEN, detail="Not an admin")
    return identity
```

And using this function as a dependency instead of `get_identity_optional`.

Reply:
I've implemented this suggestion in [95cfbfb](https://github.com/fedora-infra/fmn/pull/748/commits/95cfbfbc6f3b8be55fe3a5dd92cf8105a2971f41).